tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersQualifiedNames.cs

```diff
@@ -0,0 +1,108 @@
+                new MongoDB.Driver.FindOptions() { MaxTime = System.TimeSpan.FromSeconds(3), Max = new MongoDB.Bson.BsonDocument("min", 2) });
+        }
+
+        [BuildersMQL("{ \"field\" : { \"$elemMatch\" : { \"field\" : fieldValue } } }")]
+        [BuildersMQL("{ \"ByteNullable\" : MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyByteNullable }")]
+        [BuildersMQL("{ \"$or\" : [{ \"SiblingsCount\" : MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyByte }, { \"SiblingsCount\" : MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyShort }, { \"SiblingsCount\" : MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyInt }, { \"TicksSinceBirth\" : NumberLong(MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyLong) }, { \"Name\" : MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyString }, { \"Name\" : MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyString }, { \"Name\" : MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyString2 }, { \"Name\" : MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyString }, { \"Vehicle.VehicleType.Type\" : MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyEnum }, { \"Vehicle.VehicleType.MPG\" : MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyDouble }] }")]
+        public void Qualified_inside_method()
+        {
+            var fieldValue = "fieldValue";
+            _ = Builders<BsonDocument>.Filter.ElemMatch("field", MongoDB.Driver.Builders<MongoDB.Bson.BsonValue>.Filter.Eq("field", fieldValue));
+
+            _ = Builders<NullableHolder>.Filter.Eq(n => n.ByteNullable, MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyByteNullable);
+
+            _ = Builders<Person>.Filter.Eq(p => p.SiblingsCount, MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyByte) |
+                Builders<Person>.Filter.Eq(p => p.SiblingsCount, MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyShort) |
+                Builders<Person>.Filter.Eq(p => p.SiblingsCount, MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyInt) |
+                Builders<Person>.Filter.Eq(p => p.TicksSinceBirth, MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyLong) |
+                Builders<Person>.Filter.Eq(p => p.Name, MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyString) |
+                Builders<Person>.Filter.Eq(p => p.Name, MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyString) |
+                Builders<Person>.Filter.Eq(p => p.Name, MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyString2) |
+                Builders<Person>.Filter.Eq(p => p.Name, MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyString) |
+                Builders<Person>.Filter.Eq(p => p.Vehicle.VehicleType.Type, MongoDB.Analyzer.Tests.Common.DataModel.StaticHolder.ReadonlyEnum) |
```

Comment:
Same naming pattern, no need to test all types.
Also consider combing Qualified_inside_method and Qualified_inside_namespace into single test with different naming nesting levels. Given `A.B...X.Y.Class.Property` is a full name, let's test:
```
A.B...X.Y.Class.Propert
B...X.Y.Class.Propert
X.Y.Class.Propert
Y.Class.Propert
```
Same for `Builders<A.B...X.Y.Class>`.

Also  test cases like:

```
using user = MongoDB.Analyzer.Tests.Common.DataModel.User;
using driver = MongoDB.Driver;
_ = driver.Builders<user>.Filter.Eq(user => user.Age, 22);
_ = driver.Builders<MongoDB.Analyzer.Tests.Common.DataModel.User>.Filter.Eq(user => user.Age, 22);
```

Reply:
Done